### DISCUSSION TITLE
Reference haplotype in imputed hVCF files

### DISCUSSION THREAD

**Discussion by beantkapoor786** [maize-genetics/phg_v2#228] [Q&A]
Why don't the hVCF files created after `find-paths` command using short-read data have reference haplotype paths in them?

**Comment by zrm22** [comment]
Was the reference hvcf included in the hvcf-dir used for the build-kmer-index step?  If you are just using the hvcf dir created from the  create-maf-vcf command it will not have the reference included.

**Comment by beantkapoor786** [comment]
So, my hvcf-dir doesn't have the reference hvcf file. But, shouldn't it have the file as default output? What steps can I do now to include it in the database so that the imputation reflects haplotype paths from reference file?

**Reply by lynnjo** [reply]
@beantkapoor786 The reference hvcf file is created when you run create-ref-vcf.  At this time, the reference hvcf file is put into a folder named "hvcf_files" which is a subfolder of your db-path folder.  This step happens prior to running align-assemblies or create-maf-vcf

When you run create-maf-vcf, you specify an output folder to put the g.vcf and h.vcf files that were created.  The ref h.vcf is not processed by create-maf-vcf so it is not stored to this folder.  When you call load-vcf, the files from the specified folder are loaded to the db.  At this time the software also puts a copy of the h.vcf files into the same folder where the reference was stored (your <db-path>/hvcf_files.

If you use that folder, you'll have all the hvcf files, including the reference.  If you use the folder where create-maf-vcf stores the h.vcf files, you will be missing the reference because create-maf-vcf does not process the reference.

In any case, the reference is in the db, you just didn't pick it up because you gave build-kmer-index an hvcf-dir that did not include the reference.

**Comment by zrm22** [comment]
The reference hvcf is in the DB it just is not included when you run the create-maf-vcf command because it is already there prior to running this step.

You can either run the export-vcf step or you can copy the reference h.vcf into that directory.  Then rerun your build-kmer-index, map-kmer and find-paths steps and they should be represented.
